toolchain/check/impl_lookup.cpp

```diff
@@ -211,9 +211,14 @@ static auto GetInterfacesFromConstantId(
   if (!identified_id.has_value()) {
     return std::nullopt;
   }
-  return {{.interfaces = context.identified_facet_types()
-                             .Get(identified_id)
-                             .required_interfaces(),
+  // TODO: IdentifiedFacetTypes are held in a RelationalValueStore which does
+  // not protect against UAF through reallocation, so we must copy data out of
+  // it.
+  llvm::SmallVector<SemIR::SpecificInterface> interfaces(
+      context.identified_facet_types()
+          .Get(identified_id)
+          .required_interfaces());
+  return {{.interfaces = std::move(interfaces),
```

Comment:
`SmallVector` moves aren't especially cheap due to needing to copy the inline storage; maybe construct this in-place in `.interfaces` instead to avoid one of the two copies?
```suggestion
  // TODO: IdentifiedFacetTypes are held in a RelationalValueStore which does
  // not protect against UAF through reallocation, so we must copy data out of
  // it.
  return {{.interfaces = context.identified_facet_types()
               .Get(identified_id)
               .required_interfaces(),
```

Reply:
This change is reverted in this PR now that #6399 landed.